app/src/main/java/dev/five_star/mybooks/database/BookDao.kt

```diff
@@ -0,0 +1,24 @@
+package dev.five_star.mybooks.database
+
+import androidx.room.Dao
+import androidx.room.Insert
+import androidx.room.OnConflictStrategy
+import androidx.room.Query
+import kotlinx.coroutines.flow.Flow
+
+@Dao
+interface BookDao {
+
+    @Query("SELECT * FROM book_table")
+    fun getAllBooks(): Flow<List<Book>>
+
+    @Query("SELECT * FROM book_table WHERE id = :bookId")
+    fun getBook(bookId: Int) : Book
+
+    @Insert(onConflict = OnConflictStrategy.REPLACE)
+    suspend fun insertBook(book: Book)
```

Comment:
I think we mentioned this last time, but I'll mention again just to be thorough.  There's an `@Update` annotation that we can use if/when needed.  So if you wanted, you could make your apis on this DAO more explicit by separating out the insert and update functionality into 2 separate methods.